service/config.go

```diff
@@ -32,10 +45,52 @@ func (s *configService) SetJWTSecret(secret []byte) error {
 	return s.repo.SetJWTSecret(secret)
 }
 
-// SetSuperAdmin will first make sure the provided username is a valid account
-// and then assign it as the SuperAdmin role. Only one is allowed to be set at a
-// time.
-func (s *configService) SetSuperAdmin(name string) error {
-	// TODO: Make sure this is a valid user first.
-	return s.repo.SetSuperAdmin(name)
+// CreateSuperAdmin will take a username, generate a new password, and save this
+// user with SuperAdmin permissions.
+// TODO: Refactor so we don't return a full user and pass?
+func (s *configService) CreateSuperAdmin(name string) (*model.User, string, error) {
+	sa, err := s.repo.SuperAdmin()
+	if err != nil {
+		return nil, "", err
+	}
+	if sa != "" {
+		return nil, "", ErrSuperAdminExists
+	}
+
+	p, err := newPassword()
+	if err != nil {
+		return nil, "", err
+	}
+
+	u, err := model.NewUser(name, p, auth.RoleSuperAdmin)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.userService.SaveUser(u)
+	if err != nil {
+		return nil, "", err
+	}
+
+	err = s.repo.SetSuperAdmin(name)
+	return u, p, err
+}
+
+// ResetSuperAdmin will delete the currently stored username for Super Admin,
+// which allows a new Super Admin to be initialized.
+func (s *configService) ResetSuperAdmin() error {
+	return s.repo.SetSuperAdmin("")
+}
+
+// newPassword generates 32 cryptographically secure random bytes, base64
```

Comment:
if we ever up this to 64 bits this comment is gonna be wrong, maybe removing the specifics would be a good idea?

Reply:
yeah probably